index.js

```diff
@@ -4702,9 +4703,35 @@ export async function createCsharpBom(path, options) {
         pkgData = pkgData.slice(1);
       }
       const dlist = parseCsPkgData(pkgData);
+      const deps = dlist;
       if (dlist?.length) {
         pkgList = pkgList.concat(dlist);
       }
+
+      if (dlist?.length) {
+        for (const p of dlist) {
+          parentDependsOn.add(p);
```

Comment:
Can this not store the bom-ref similar to 4723. I think we don't need another loop in 4720.

Reply:
Same as above, with these changes only i am able to generate sbom with dependencies.

Reply:
The second loop appears unnecessary. It is only checking if `p` has both name and version.

Reply:
Yes you are correct, Its a overlook. I just updated the code.